### ISSUE TITLE
trainer.fit(model, config.train_dl, config.test_dl)

### ISSUE THREAD

**Issue by pcjdhhhh** [zshhans/MSD-Mixer#8]
Hello, we noticed that the training code in the cls_exp.py file is written as trainer.fit(model, config.train_dl, config.test_dl).

Does this mean that you are using the test set as the validation set? From my understanding, this shouldn't be the case. 

I would really appreciate it if you could clarify this concern. Thank you!

**Comment by zshhans** [comment]
Hi pcjdhhhh! Thank you for your interest in our work!

It is true that for the classification experiments in this work we use the test set for validation. The reason is as follows:
1. Long story short, we adopt this experiment setting from previous works in the literature. In this work we compare our method with various task-general and task-specific methods across five tasks, with classification being one of the task. By reviewing the experimentation of existing classification methods in the literature, we find this test-set-for-validation setting was used by many, if not all, existing methods, including task-general ones (e.g., those based on the library [TSLib](https://github.com/thuml/Time-Series-Library/blob/main/exp/exp_classification.py#L81)) and task-specific ones (e.g., [FormerTime](https://github.com/icantnamemyself/FormerTime/blob/main/process.py#L91)). Therefore we adopt the same setting for fair comparison.
2. We find that the UEA benchmark datasets for TSC are usually of very small scale. The provided training sets are already so small that if we furthur split of them for validation will cause problem for the model training, and makes the result less meaningful. We guess that is the reason why many existing works used this test-set-for-validation setting in their experiments.

Hope the information above help address your concern and please feel free to ask if you have other questions.

If you find our work interesting, please consider citing our work, starring this repo, and sharing our paper and repo with others, thank you!
